### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by kal2944**
### **IN THIS PART OF THE PROJECT I HAVE UPDATED THE FOLLOWING ACTIVITIES**

- [x] Created a class as Nameable
          -  that Implement a method called correct_name that will raise a NotImplementedError

- [x] Turned my Person class to Nameable that my Person class inherits from Nameable

- [x] Prepared base Decorator that it inherits from Nameable.

- [x] Prepared CapitalizeDecorator and TrimmerDecorator that both inherits from the base Decorator class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+  attr_accessor :age
   attr_reader :id
 
   def initialize(age, parent_permission: true, name: 'Unknown')
+    super(name)
     @id = rand(1..1000)
-    @name = name
     @age = age
     @parent_permission = parent_permission
   end
```
Comment (DammyShittu) [inline person.rb]: - [ ] I was having difficulty testing your decoration. Kindly remove `name` from the super method and add the initial line you had before. You should have something like this

  ```
  def initialize(age, parent_permission: true, name: 'Unknown')
    super()
    @id = rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @kal2944,

## Highlights
✔️ Class decoration

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
   attr_reader :id
 
-  def initialize(age, parent_permission: true, name: 'Unknown')
+  def initialize(age, parent_permission: true)
+    super()
     @id = rand(1..1000)
-    @name = name
     @age = age
     @parent_permission = parent_permission
   end
```
Comment (DammyShittu) [inline person.rb]: - [ ] Kindly pay attention to the details. In your change you removed `name` from the parameters, which means we can't even create a person object at all. You also have not added the line `@name = name`. These cause errors:

  image

  Kindly ensure your code matches this:
  
  ```
  def initialize(age, parent_permission: true, name: 'Unknown')
    super()
    @id = rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @kal2944,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (kal2944) [reply person.rb]: Thank you, fixed.

File: person.rb [hunk 3]
```diff
@@ -1,18 +1,25 @@
-class Person
-  attr_accessor :name, :age
+require './nameable'
+
+class Person < Nameable
+  attr_accessor :age
   attr_reader :id
 
   def initialize(age, parent_permission: true, name: 'Unknown')
+    super()
```
Comment (codepantha) [inline person.rb]: - [ ] Please see the `ArgumentError` in the Person class constructor due to your setting `name: 'Unknown'`. The implication of this is that you can no longer instantiate the Person class while supplying the `name` attribute like this: `Person.new(22, 'shedrackstormfield')`
image

NB: the name variable must not have a default value. Use the line below to get your program running correctly 😎 
```ruby
def initialize(age, name, parent_permission: true)
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @kal2944 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @kal2944 , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- GitHub flow is used correctly. ✔
- Well done implementing the requested change. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**